### ISSUE TITLE
Graphics artifacts on Apple Silicon host (high screen tearing and mouse disappearing)

### ISSUE THREAD

**Issue by marco-04** [utmapp/UTM#2569]
**Describe the issue**
Windows 10 boots and I'm able to set it up, but there are many tearing problems, and sometimes the mouse cursor disappears, and this just in idle and by moving some windows. I have not encountered these problems by using ACVM
Here are some screenshots:
<img width="1680" alt="Schermata 2021-05-24 alle 23 25 57" src="https://user-images.githubusercontent.com/61947904/119409778-1d9a0600-bce8-11eb-80e1-6a7c28a0d0dc.png">
<img width="1680" alt="Schermata 2021-05-24 alle 23 26 08" src="https://user-images.githubusercontent.com/61947904/119409852-3aced480-bce8-11eb-8d9f-5c240a90f7c1.png">
<img width="1680" alt="Schermata 2021-05-24 alle 23 25 30" src="https://user-images.githubusercontent.com/61947904/119409939-589c3980-bce8-11eb-89ec-8d9c939a3f56.png">
![image](https://user-images.githubusercontent.com/61947904/119410036-79fd2580-bce8-11eb-8b62-92a11151f776.png)

In the last screenshot is difficult to spot, but I'm dragging my mouse around, and the selecting area is showing, but not the cursor

**Configuration**
* UTM Version: 2.1.1 beta
* OS Version: 11.3.1
* Intel or Apple Silicon? Apple Silicon

**Debug log**
[debug.log](https://github.com/utmapp/UTM/files/6535082/debug.log)

**Upload VM**
VM made with Gallery guide for windows 10 ARM

**Comment by osy** [comment]
Is this before or after installing the graphics drivers? Or both?

**Comment by marco-04** [comment]
> Is this before or after installing the graphics drivers? Or both?

Both, during configuration it wasn't that crazy, but after configuration, even without installing drivers and rebooting, this stuff happens

**Comment by marco-04** [comment]
Also I forgot to mention that I've tried also previous versions of UTM, but I have the same issue

**Comment by muratdk23** [comment]
Same issue on all versions.

**Comment by iRonJ** [comment]
I  see this same exact artifact, but only  when trying to watch a video. Using a standalone qemu, I don’t see this issue. I’ve linked my qemu settings in #2557

**Comment by Morac2** [comment]
I see the same problem.  It seems to be tied to using virtio-ramfb display card.  Using ramfb is better, though not perfect.

Of note using ACVM with the virtio-gpu drivers doesn't appear to have that problem.  Though I can't adjust the resolution as the display settings immediately close with those drivers.  See https://github.com/ubenmackin/ACVM

**Comment by osy** [comment]
@marco-04 @Morac2 do you guys only see this in v2.1.1 beta or do you also see it in v2.0.28?

**Comment by marco-04** [comment]
> @marco-04 @Morac2 do you guys only see this in v2.1.1 beta or do you also see it in v2.0.28?

I tried also older versions and they all have the same problem. I don't remember which versions I tried exactly (I believe they  were 2.0.15, 2.0.16, 2.0.17, 2.0.21, 2.0.22, 2.0.23, 2.0.24 and 2.0.25), but I had the same tearing issue

**Comment by themak1985** [comment]
Same issue here - using 2.1.1 - the tearing isnt as bad, but the mouse is always dissapearing.

**Comment by Samyak-Coder** [comment]
Hey can you tell me how to get macos on UTM latest version on iPadOS 14.7? It’s quite confusing

**Comment by litori** [comment]
Can confirm screen tearing on current and previous versions of UTM and MacOS since 11.2.  I just assumed that this was due to UTM not fully mature for Apple silicon.  Mouse disappearing is definitely a very big problem for me personally.

**Comment by conath** [comment]
> […] I have not encountered these problems by using ACVM

Since ACVM uses "ramfb" video device, have you tried changing the Display configuration to that?
<img width="803" alt="UTM VM configuration page for Display settings, with the dropdown for Emulated Display Card selected and the item RAM framebuffer standalone device highlighted" src="https://user-images.githubusercontent.com/12073163/120803489-e4616180-c543-11eb-9f95-cf4e5a251232.png">
 
By the way, when using ramfb you will have low resolution, unless you use custom EFI (you want `QEMU_EFI-AARCH64-HighRes-UTM-logo.fd`, import as "BIOS" or "Disk Image" with "PC System Flash" interface): https://chrisp.cafe/OVMF/ 

> **Debug log**
> [debug.log](https://github.com/utmapp/UTM/files/6535082/debug.log)

This indicates you were using `virtio-ramfb`. No drivers available for that. Install this first: https://github.com/utmapp/qemu/releases/download/v5.2.0-asi/spice-guest-tools-0.164.iso
Then switch to `virtio-gpu-pci` in the Display configuration for driver support with this specific driver installer. Your VM will show "guest disabled display" while it's booting, don't worry. Just wait. It has 2D and some software 3D acceleration.

**Comment by marco-04** [comment]
> > […] I have not encountered these problems by using ACVM
> 
> Since ACVM uses "ramfb" video device, have you tried changing the Display configuration to that?
> <img alt="UTM VM configuration page for Display settings, with the dropdown for Emulated Display Card selected and the item RAM framebuffer standalone device highlighted" width="803" src="https://user-images.githubusercontent.com/12073163/120803489-e4616180-c543-11eb-9f95-cf4e5a251232.png">
> 
> By the way, when using ramfb you will have low resolution, unless you use custom EFI (you want `QEMU_EFI-AARCH64-HighRes-UTM-logo.fd`, import as "BIOS" or "Disk Image" with "PC System Flash" interface): https://chrisp.cafe/OVMF/
> 
> > **Debug log**
> > [debug.log](https://github.com/utmapp/UTM/files/6535082/debug.log)
> 
> This indicates you were using `virtio-ramfb`. No drivers available for that. Install this first: https://github.com/utmapp/qemu/releases/download/v5.2.0-asi/spice-guest-tools-0.164.iso
> Then switch to `virtio-gpu-pci` in the Display configuration for driver support with this specific driver installer. Your VM will show "guest disabled display" while it's booting, don't worry. Just wait. It has 2D and some software 3D acceleration.

Yes I tried to switch the graphics, but I think it didn't even show the booting screen. Anyway I'm going to try both solutions today.

**Comment by marco-04** [comment]
> Since ACVM uses "ramfb" video device, have you tried changing the Display configuration to that?

<img width="1409" alt="Schermata 2021-06-04 alle 15 05 21" src="https://user-images.githubusercontent.com/61947904/120805781-75393c80-c546-11eb-9983-4143dca16734.png">

This is what happens when I switch to RAMFB, it didn't even show the TianoCore booting screen

**Comment by conath** [comment]
I don't have Apple Silicon Mac to test so my results may differ because of that. Mine shows the boot screen fine, however I am using my custom EFI. (you want QEMU_EFI-AARCH64-HighRes-UTM-logo.fd, import as "BIOS" or "Disk Image" with "PC System Flash" interface): https://chrisp.cafe/OVMF/

Note that that file is going to overall reduce performance, as it was compiled in Debug mode. It's not intended as a permanent solution.

**Comment by marco-04** [comment]
Ok, I tried and it is way better than virtio-ramfb. I will still try the other method you mentioned, to see if it works better

**Comment by marco-04** [comment]
Ok, just tried the other method and it still has tearing and disappearing mouse. Only RAMFB has the lowest amount of tearing and the mouse does not disappear

**Comment by hiatsu0** [comment]
Can confirm this issue on M1, OS 11.4, UMT 2.1.1 (had this also with 2.0.27), virtio-ramfb, Windows 10 ARM64 Build 21354.

Also noticed that the cursor icon colors are _inverted_ (blue loading animation in cursor is orange, the cursor itself is still white). For now the best workaround I've come up with is to Port forward :3389->:3389 and use Microsoft Remote Desktop app to RDP locally into the virtual machine.

**Comment by volisoft** [comment]
Not sure if this is Windows-specific, because this also happens when running Ubuntu 20.04 ARM64 guest on Apple M1, OS 11.4, UTM 2.1.1 (and older versions of UTM).

**Comment by sandeepwadhwa-ai** [comment]
I am having similar problem on MacBook Air (Big Sur 11.4) for Ubuntu 20.04 ARM 64 as guest.

**Comment by superkid200** [comment]
This issue seems to be universal across all virtual machines.

**Comment by thomasfchang** [comment]
Having the same issue when using a Debian based VM. MacOS 11.4, UTM v2.1.1 , Apple Silicon. Tried to install the SPICE tools and switching from virtio-ramfb to virtio-gpu-pci. I don't have the issue with bare QEMU. It is not completely fluent then but there is no tearing and it taking seconds to images the window when it has moved position. I applied installed this patch for normal QEMU, maybe it will work when incorporating it with UTM: https://github.com/knazarov/homebrew-qemu-virgl
